File: client/src/app/modules/miscellaneous/services/phone.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/appRest.service';
import { Phone } from '../../tiers/model/Phone';

@Injectable({
  providedIn: 'root'
})
export class PhoneService extends AppRestService<Phone>{

  constructor(http: HttpClient) {
    super(http, "tiers");
  }

  getPhones(phone: string) {
    return this.getList(new HttpParams().set("phone", phone), "phones/search");
  }

}

File: src/main/java/com/jss/jssbackend/modules/tiers/service/PhoneService.java
package com.jss.jssbackend.modules.tiers.service;

import java.util.List;

import com.jss.jssbackend.modules.tiers.model.Phone;

public interface PhoneService {
    public List<Phone> findPhones(String phone);

    public Phone getPhone(Integer id);

    public void populateMPhoneIds(List<Phone> phones);
}
